src/NuGet.Core/NuGet.Build.Tasks/TaskLoggingQueue.cs

```diff
@@ -19,13 +20,18 @@ internal class TaskLoggingQueue : LoggingQueue<string>
         /// </summary>
         private readonly TaskLoggingHelper _log;
 
+        private readonly ICollection<string> _restoredProjectOutputPaths;
+
         /// <summary>
         /// Initializes a new instance of the TaskLoggingHelperQueue class.
         /// </summary>
         /// <param name="taskLoggingHelper">The <see cref="Task" /> to create a logging queue for.</param>
-        public TaskLoggingQueue(TaskLoggingHelper taskLoggingHelper)
+        /// <param name="restoredProjectOutputPaths">Receives the list of output paths for the restored projects.</param>
```

Comment:
This is often (always) an empty list? The dequeuing of a task can modify the list, right? Currently sounds like it's supposed to be a complete list of paths.

If it's just always initializing the list, maybe it shouldn't be a constructor parameter?

```suggestion
        /// <param name="restoredProjectOutputPaths">Receives the list of initial output paths (can be empty) for the restored projects. </param>
```